modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java

```diff
@@ -456,7 +458,11 @@ private void onEvent(CoreEvent event, ExecutorCallback executorCallback, Context
       OperationExecutionFunction operationExecutionFunction = (parameters, operationEvent, callback) -> {
         setOperationExecutionParams(location, event, configuration, parameters, operationEvent, callback, ctx);
 
-        fluxSupplier.get().next(operationEvent);
+        if (fluxSupplier instanceof TransactionAwareFluxSinkSupplier && TransactionUtils.isTxActive(ctx)) {
+          ((TransactionAwareFluxSinkSupplier<CoreEvent>) fluxSupplier).getThreadFluxSink().next(operationEvent);
+        } else {
+          fluxSupplier.get().next(operationEvent);
+        }
```

Comment:
can this logic be delegated to the `fluxSupplier` instead of being here? (essentially get rid of the `instanceof`)